pkg/nodeps/utils_test.go

```diff
@@ -0,0 +1,85 @@
+package nodeps
+
+import (
+	"context"
+	"errors"
+	"github.com/stretchr/testify/assert"
+	"net"
+	"testing"
+	"time"
+)
+
+type netResolverStub struct {
+	sleepTime time.Duration
+	err       error
+}
+
+func (t netResolverStub) LookupHost(ctx context.Context, _ string) ([]string, error) {
+	select {
+	case <-time.After(t.sleepTime):
+	case <-ctx.Done():
+		return nil, errors.New("context timed out")
+	}
+	return nil, t.err
+}
+
+func resetVariables() {
+	// resetVariables back to defaults
+	isInternetActiveNetResolver = net.DefaultResolver
+	isInternetActiveAlreadyChecked = false
+	isInternetActiveResult = false
+}
+
+func TestIsInternetActiveErrorOccurred(t *testing.T) {
+	resetVariables()
+
+	isInternetActiveNetResolver = netResolverStub{
+		sleepTime: 0,
+		err:       errors.New("test error"),
+	}
+
+	assert.False(t, IsInternetActive())
+}
+
+func TestIsInternetActiveTimeout(t *testing.T) {
+	resetVariables()
+
+	isInternetActiveNetResolver = netResolverStub{
+		sleepTime: 501 * time.Millisecond,
```

Comment:
```suggestion
		sleepTime: 1000 * time.Millisecond,
```

Let's set sleepTime in TestIsInternatActiveTimeout dramatically higher, like 1000 * time.Millisecond. There's no reason to have 501 when 500 is the set difference. The possibility of a 1ms deficiency in resolution is high. I'd say that's probably the first thing to do, unless I'm misunderstanding the code.

Reply:
Yeah, thats a good idea